### ISSUE TITLE
Check asymmetry of losses (detect negative edge)

### ISSUE THREAD

**Issue by deanturpin** [deanturpin/lft#24]
## Problem

**"Picking up pennies in front of a steamroller"** - strategies that win often but lose catastrophically.

Many small wins + occasional big losses = hidden negative edge.

### The Trap

Imagine:
- **9 winning trades**: +$10 each = +$90
- **1 losing trade**: -$100 = -$100
- **Win rate**: 90% (looks great!)
- **Net P&L**: -$10 (actually losing money!)

This pattern is common when:
- Selling options premium (consistent small wins, rare massive losses)
- Mean reversion without stops (keeps adding to losers)
- Tight take profits with wide stops (win small, lose big)

### Key Metrics

**1. Profit Factor** = Total Wins / Total Losses
- \> 2.0 = Excellent (winning twice as much as losing)
- \> 1.5 = Good (sustainable edge)
- < 1.5 = Warning (barely profitable)
- < 1.0 = Losing strategy

**2. Payoff Ratio** = Average Win / Average Loss
- \> 1.0 = Wins bigger than losses (healthy)
- < 1.0 = Losses bigger than wins (dangerous asymmetry)
- < 0.5 = Severe asymmetry (red flag)

**3. Tail Risk** = Largest Loss / Average Loss
- < 2.0 = Normal distribution
- 2.0-3.0 = Some outliers
- \> 5.0 = Fat tail risk (occasional huge losses)

### Real Example

**Strategy A (Healthy):**
```
Win rate: 50%
Avg win: $50, largest: $75
Avg loss: $30, largest: $45
Profit factor: 1.67
Payoff ratio: 1.67
Tail risk: 1.5x
✅ Balanced risk/reward
```

**Strategy B (Hidden Danger):**
```
Win rate: 80%
Avg win: $10, largest: $15
Avg loss: $40, largest: $200
Profit factor: 0.5
Payoff ratio: 0.25
Tail risk: 5.0x
❌ "Pennies in front of steamroller"
```

### Phase 1: Track Statistics

Add to backtest stats:

```cpp
struct StrategyStats {
  // Existing fields
  std::string name;
  int total_trades;
  double total_pnl;
  double win_rate;
  
  // New fields for asymmetry analysis
  double total_wins{0.0};
  double total_losses{0.0};
  int wins_count{0};
  int losses_count{0};
  double largest_win{0.0};
  double largest_loss{0.0};
  
  // Calculated metrics
  double profit_factor() const {
    return total_losses != 0.0 ? total_wins / std::abs(total_losses) : 0.0;
  }
  
  double payoff_ratio() const {
    auto avg_win = wins_count > 0 ? total_wins / wins_count : 0.0;
    auto avg_loss = losses_count > 0 ? total_losses / losses_count : 0.0;
    return avg_loss != 0.0 ? avg_win / std::abs(avg_loss) : 0.0;
  }
  
  double tail_risk() const {
    auto avg_loss = losses_count > 0 ? total_losses / losses_count : 0.0;
    return avg_loss != 0.0 ? std::abs(largest_loss / avg_loss) : 0.0;
  }
};
```

### Phase 2: Enhanced Backtest Output

```
✓ ma_crossover Complete: 228 trades, $702.98 P&L, 43.0% WR, 34.5 bps avg
  
  📊 Win/Loss Distribution:
  Wins:   98 trades (43.0%), avg +$5.20, largest +$15.30
  Losses: 130 trades (57.0%), avg -$2.80, largest -$8.50
  
  📈 Risk Metrics:
  Profit Factor: 1.82 ✅ (1.5+ is good)
  Payoff Ratio:  1.86 ✅ (wins 1.86x bigger than losses)
  Tail Risk:     3.04 ⚠️  (largest loss 3x average - monitor)
  
  ✅ Healthy strategy - no asymmetry detected
```

### Phase 3: Warning Flags

Display warnings for dangerous patterns:

```
✓ volume_surge Complete: 132 trades, $358.49 P&L, 43.9% WR
  
  ⚠️  WARNING: Asymmetric loss distribution detected
  
  Profit Factor: 1.05 ⚠️  (barely profitable)
  Payoff Ratio:  0.60 ❌ (losses 67% bigger than wins)
  Tail Risk:     6.2x 🚨 (fat tail - occasional huge losses)
  
  Analysis: This strategy wins often but loses big.
  Risk: One bad trade could wipe out weeks of gains.
  Recommendation: Review stop loss logic or disable strategy.
```

## Red Flags to Detect

**Auto-disable strategies if:**
- [ ] Profit factor < 1.0 (losing money overall)
- [ ] Payoff ratio < 0.5 (losses 2x bigger than wins)
- [ ] Tail risk > 10.0 (catastrophic outlier losses)
- [ ] Win rate > 70% but profit factor < 1.5 (classic trap pattern)

**Warn about strategies with:**
- [ ] Profit factor 1.0-1.5 (marginal edge)
- [ ] Payoff ratio < 1.0 (negative asymmetry)
- [ ] Tail risk 5.0-10.0 (significant outliers)

## Why This Matters for LFT

From your 15-minute bar backtests:
- **ma_crossover**: $702.98 P&L, 43.0% WR, 34.5 bps avg
- **mean_reversion**: $332.06 P&L, 44.5% WR, 46.1 bps avg

**Questions to answer:**
- Are those losses evenly distributed?
- Or are there occasional -200 bps losses dwarfing +46 bps wins?
- Is the 43% win rate sustainable?
- Are you one catastrophic trade away from giving it all back?

**Example concern:**
If mean_reversion has:
- 57 small wins averaging +$10 (total +$570)
- 71 small losses averaging -$2 (total -$142)
- But 1 huge loss of -$100

Then: Win rate 44.5% ✅, Total P&L $328 ✅, **but hidden tail risk** 🚨

## Priority

**Medium** - Important for risk management, but need data first

**Prerequisites:**
1. ✅ 1+ month of live trading data
2. ✅ Trade duration tracking (#33 complete)
3. ✅ Baseline performance established

**Timeline:** February 2026 after accumulating sufficient trade data

## Success Criteria

- [ ] Backtest output shows profit factor, payoff ratio, tail risk
- [ ] Visual distinction between healthy and dangerous strategies
- [ ] Auto-disable strategies with catastrophic asymmetry
- [ ] Warnings for marginal strategies
- [ ] Can identify "pennies in front of steamroller" patterns

## Related Issues

- Depends on #33 (trade duration) - need full trade data
- Related to #37 (adaptive exits) - could help reduce tail risk
- Related to #19 (cooldowns) - prevent revenge trading after big losses

## References

- Classic example: Long-Term Capital Management (LTCM) collapse 1998
- Nassim Taleb's "Black Swan" - tail risk in financial markets
- Options sellers' biggest fear: unlimited downside risk

## Notes

The quote "picking up pennies in front of a steamroller" perfectly captures the danger: you win consistently (picking up pennies) until a rare catastrophic event (steamroller) wipes out all gains and more.

This analysis protects against that exact scenario.
